src/ParseObject.js

```diff
@@ -400,7 +400,18 @@ class ParseObject {
       } else if (attr === 'ACL') {
         changes[attr] = new ParseACL(response[attr]);
       } else if (attr !== 'objectId') {
-        changes[attr] = decode(response[attr]);
+        const val = decode(response[attr]);
+        if (val && typeof val === Object
```

Comment:
typeof val === 'object' should fix the tests

Reply:
@dplewis Thanks for taking a look.  Unfortunately it does not.  

Interestingly, the code as it is written only fails on the new tests added.  However, changing it to `typeof val === 'object'` per your suggestion results in 2 failures in legacy unit tests.

```
 FAIL  src/__tests__/ParseObject-test.js
  ● ParseObject › commits changes to server data when saved

    expect(received).toEqual(expected) // deep equality

    - Expected
    + Received

      Object {
        "age": 24,
    -   "updatedAt": 2020-09-14T12:57:56.944Z,
    +   "updatedAt": Object {},
      }

      1010 |       updatedAt: { __type: 'Date', iso: updated.toISOString() }
      1011 |     });
    > 1012 |     expect(p._getServerData()).toEqual({
           |                                ^
      1013 |       updatedAt: updated,
      1014 |       age: 24
      1015 |     });

      at Object.it (src/__tests__/ParseObject-test.js:1012:32)

  ● ParseObject › interpolates delete operations

    expect(received).toBeUndefined()

    Received: {}

      1289 |     o.save({ key: 'value', deletedKey: 'keyToDelete' }).then(() => {
      1290 |       expect(o.get('key')).toBe('value');
    > 1291 |       expect(o.get('deletedKey')).toBeUndefined();
           |                                   ^
      1292 |       done();
      1293 |     });
      1294 |   });

      at o.save.then (src/__tests__/ParseObject-test.js:1291:35)

```

Reply:
At least your test cases passed. I can look into the other tests.

Reply:
type `object` can be many fields. These tests are failing for type `Date` and `Unset`. Look into `decode.js` for additional types.

Try 

```
if (val && typeof val === 'object'
          && !(val instanceof Array)
          && !(val instanceof ParseObject)
          && !(val instanceof ParseFile)
          && !(val instanceof ParseRelation)
          && !(val instanceof Op)
          && !(Object.prototype.toString.call(val) === '[object Date]')) {
```

Reply:
Might have to add check for `ParseGeoPoint` and `ParsePolygon` and test cases for those.

Reply:
> Might have to add check for `ParseGeoPoint` and `ParsePolygon` and test cases for those.

@dplewis Ok, thanks... will check that out in a bit.

The other changes did result in passing unit tests.  However, I am still failing the integration test I created.  I really don't understand how considering that I have seen it work as expected when actually used in practice.  The test is super simple...  The issue is the local value of _obj.objectField_ AFTER the **save()**.  I really wonder if the cleaner fix to this issue would be for the **save()** to return the entire object even if only part of it changed.  Thoughts?

```
it('can increment nested field and retain full object', async () => {
    const obj = new TestObject();
    obj.set('objectField', { number: 5, letter: 'a' });
    assert.equal(obj.get('objectField').number, 5);
    assert.equal(obj.get('objectField').letter, 'a');
    await obj.save();

    obj.increment('objectField.number', 15);
    assert.equal(obj.get('objectField').number, 20);
    assert.equal(obj.get('objectField').letter, 'a');
    await obj.save();

    assert.equal(obj.get('objectField').number, 20);
    assert.equal(obj.get('objectField').letter, 'a');

    const query = new Parse.Query(TestObject);
    const result = await query.get(obj.id);
    assert.equal(result.get('objectField').number, 20);
    assert.equal(result.get('objectField').letter, 'a');
  });

```

Reply:
Are you running `npm run watch:node`? On the Travis build the test you created passed.